.chloggen/routing-processor-rename-config-parameter.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
I did it as a breaking change without deprecating the `expression` config since we declared OTTL support in the routing processor as experimental/tech preview. Let me know if you think that `expression` has to be deprecated first, I'll update PR then.